rust/numaflow-core/src/config.rs

```diff
@@ -42,20 +42,63 @@ pub(crate) struct Settings {
     pub(crate) custom_resource_type: CustomResourceType,
 }
 
+/// LookbackStruct is a struct to hold the lookback window map for the monovertex/pipeline.
+#[derive(Default)]
+pub(crate) struct LookbackStruct {
+    window_map: parking_lot::Mutex<[(&'static str, i64); 4]>,
+}
+
+impl LookbackStruct {
+    fn new() -> Self {
```

Comment:
`fun new(defaultLookback: i64) -> Self`?

Reply:
I was exploring that as well, just one issue that I observed was in the testing flow. Where because the config is not available its difficult to different for the call here what to pass
https://github.com/numaproj/numaflow/pull/2232/files#diff-4057e535d3ca33f1b08e56689bfca36f9ce5b7845bc1a6915aa3d343f0aefa5aR748